OpenRA.Mods.Common/Activities/Move/MoveAdjacentTo.cs

```diff
@@ -45,8 +44,9 @@ public MoveAdjacentTo(Actor self, in Target target, WPos? initialTargetPosition
 		{
 			this.target = target;
 			this.targetLineColor = targetLineColor;
-			Mobile = self.Trait<Mobile>();
-			pathFinder = self.World.WorldActor.Trait<IPathFinder>();
+
+			// PERF: Because we can be sure that OccupiesSpace is Mobile here, we can save some performance by avoiding querying for the trait.
+			Mobile = (Mobile)self.OccupiesSpace;
```

Comment:
Mods won't be able to add multiple Mobile traits to an actor? In the past it would cause an exception. Now it would cause OccupiesSpace to be used.

Reply:
Not sure what you mean? Actors are only allowed to have one of these traits, and Actor.OccupiesSpace is just `IOccupySpace` cached by the actor, which is implemented by Mobile, so in a way we're already caching Mobile and can use this casting to save a look-up.

Reply:
That was my question. Can actors only have one of these traits.

Reply:
Actors can only have one of these traits currently, and many things will need to be reworked before that can change.